lib/version.js

```
@@ -88,8 +89,82 @@ function setCommand(version, pkg) {
+      var value = scriptPhases['postinstall'].toString() + ' && node scripts/sl-blip.js';
+      project.scripts('postinstall', value);
+    }
+    updateBlipScript(project);
+  }
+  if (opted && blipScript) { //when package opted for blip & have blip in scripts, update sl-blip.js file
+    updateBlipScript(project);
+  }
+}
+
+function hasOpted(project, scriptPhases, availPhases, opted, blipScript, cb) {
+  //check if optionalDependencies has sl-blip, if yes, delete sl-blip
   if (project.optionalDep('sl-blip')) {
-    var blipUrl = 'http://blip.strongloop.com/' + project.nameVer();
-    project.optionalDep('sl-blip', blipUrl);
+    project.optionalDep('sl-blip', '');
+    opted = true;
+  }
+  //check if scripts has sl-blip setup
+  for (var phase in scriptPhases) {
+    if (scriptPhases.hasOwnProperty(phase) && !scriptPhases[phase]) {
+      availPhases.push(phase);
+    } else {
+      hasBlipScript(scriptPhases[phase]) ? opted = blipScript = true : opted;
+    }
+  }
+  cb(opted, blipScript);
+}
+
+function hasBlipScript(phaseVal) {
+  var reg = /sl-blip/i;
+  if (reg.test(phaseVal.toString())) {
+    return true;
+  }
+  return false;
+}
+
+function updateBlipScript(project) {
+  var SCRIPTS = path.resolve(project.rootPath, 'scripts'); //path to scripts dir of project
+  var SL_BLIP = path.resolve(SCRIPTS, 'sl-blip.js'); //path to sl-blip.js of project
+  var blipjs = fs.readFileSync(path.resolve(__dirname, 'sl-blip.js'), 'utf-8'); //latest sl-blip.js from strong-tools/lib/
+  
+  //if scripts dir doesn't exist, create one
+  try {
+    fs.mkdirSync(SCRIPTS);
+  } catch (e) {
+    if (e.code === 'EEXIST') {
+      //ignore error if dir exists and create blip.js file
+    } else {
+      console.error('Failed to create directory: ', e)
+    }
+  }
+  // write sl-blip.js file
+  try {
+    fs.writeFileSync(SL_BLIP, blipjs, 'utf-8');
+  } catch (e) {
+    console.error('Failed to write files: ', e)
```

While it appears to be helpful to catch the exception here and print an error message, this is suppressing what would should actually be a fatal error. It would be best to not catch this at all, unless you are going to print additional information that node won't already print on failure - and if you do you should either re-throw the exception or do `process.exit(1);`